worldedit-core/src/main/java/com/sk89q/worldedit/extension/input/ParserContext.java

```diff
@@ -90,6 +93,10 @@ public void setExtent(@Nullable Extent extent) {
         this.extent = extent;
     }
 
+    public void setLegacyMapper(@Nullable LegacyMapper legacyMapper) {
+        this.legacyMapper = legacyMapper;
+    }
+
```

Comment:
I would prefer not to make it possible to set the LegacyMapper public because I know that third party plugins will try to use it and I am not willing to support this behavior.

Reply:
The standard access modifiers aren't going to help here due to package structure.  The other approach I considered was setting the `LegacyMapper.INSTANCE` variable inside the `LegacyMapper` constructor _before_ invoking `loadFromResource`. This would ensure that any calls within the same stack to `LegacyMapper.getInstance()` would not attempt to reinitialize, but would return the same (partially constructed) instance.

Reply:
I didn't really look at the issue yet, but it didn't appear previously - I'd rather find out *why* it's happening now the way it is instead of building a workaround that is scattered all over the code base.

Reply:
I'm going to close this PR, since this approach isn't going to work (due to exposing `setLegacyMapper` publicly).  After removing the `FaweVersion` changes, it would be easier to just start fresh.  I posted another potential solution, but TBH, I don't have the understanding of the project to know what approach is best.